README.md

```diff
@@ -998,11 +999,292 @@ If you found this useful, please ⭐️and _share_ the GitHub repo
 so we know you like it!
 
 
+<br /><br /><br />
+
+## Future Steps
+### Moving state out of the LiveViews
+
+With this implementation you may have noticed that when we open a new browser 
+window the count is always zero. As soon as we click plus or minus it adjusts
+and all the views get back in line. This is because the state is replicated 
+across LiveView instances and coordinated via pub-sub.  If the state was big 
+and complicated this would get wasteful in resources and hard to manage.
+
+Generally it is good practice to identify *shared state* and to manage that in 
+a single location.
+
+The Elixir way of managing state is the 
+[GenServer](https://hexdocs.pm/elixir/GenServer.html), using PubSub to update 
+the LiveViews about changes.  This allows the LiveViews to focus on user specific
+state, separating concerns; making the application both more efficient 
+(hopefully) and easier to reason about and debug.
+
+We are now going to start maknig use of the lib/live_view_counter directory! The 
+[Phoenix docs](https://hexdocs.pm/phoenix/directory_structure.html#content) says
+that this holds "all of your business domain".  For us this is the current count, 
+along with the incr and decr methods.
+
+``` elixir
+defmodule LiveViewCounter.Count do
+  use GenServer
+
+  alias Phoenix.PubSub
+
+  @name :count_server
+
+  @start_value 0
+
+  # -------  External API (runs in client process) -------
+
+  def topic do
+    "count"
+  end
+
+  def start_link(_opts) do
+    GenServer.start_link(__MODULE__, @start_value, name: @name)
+  end
+  
+  def incr() do
+    GenServer.call @name, :incr
+  end
+
+  def decr() do
+    GenServer.call @name, :decr
+  end
+
+  def current() do
+    GenServer.call @name, :current
+  end
+
+  def init(start_count) do
+    {:ok, start_count}
+  end
+
+  # -------  Implementation  (Runs in GenServer process) -------
+
+  def handle_call(:current, _from, count) do
+     {:reply, count, count}
+  end
+
+  def handle_call(:incr, _from, count) do
+    make_change(count, +1)
+  end
+
+  def handle_call(:decr, _from, count) do
+    make_change(count, -1)
+  end
+
+  defp make_change(count, change) do
+    new_count = count + change
+    PubSub.broadcast(LiveViewCounter.PubSub, topic(), {:count, new_count})
+    {:reply, new_count, new_count}
+  end
+end
+```
+
+The GenServer runs in its own process.  Other parts of the application invoke 
+the API in their own process, these calls are forwarded to the `handle_call` 
+functions in the GenServer process where they are processed serially.  
+
+We have also moved the PubSub publication here as well.
+
+*[We could have used asyncronous `handle_cast` functions and relied on the 
+PubSub to update us.  Using `handle_call` means the calling LiveView will be 
+updated twice, but it doesn't really matter at this scale.]*
```

Comment:
it might matter if the person clicks the inc/dec buttons in the UI quite fast and the UI
receives the update to the previous  state multiple time. 💭